cmd/helmpush/main.go

```diff
@@ -335,6 +344,24 @@ func getChartmuseumError(b []byte, code int) error {
 	return fmt.Errorf("%d: %s", code, er.Error)
 }
 
+func getIndexDownloader(client *cm.Client) helm.IndexDownloader {
+	return func() ([]byte, error) {
+		resp, err := client.DownloadFile("index.yaml")
+		if err != nil {
+			return nil, err
+		}
+		defer resp.Body.Close()
+		b, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
+		if resp.StatusCode != 200 {
```

Comment:
This can be before reading body

Reply:
Let's follow the original design. The function call to `getChartmuseumError(b, resp.StatusCode)` requires reading the response body.